argoproj/argo-cd
# Adds support for Kustomize 2.0.1. Fixes #1085

This assumes that you have two binaries on your PATH:

* `kustomize` this should be version 1.0.11
* `kustomize2` this should be version 2.0.1

# Testing Locally
If you upgrade brew, you may find you delete one of them to fix-up brew:

* Make sure you have a kustomize-1.0.11.zip from Slack and copy that as /usr/local/Cellar/kustomize/1.0.11
* Make sure you have a kustomize-2.0.1.zip from Slack and copy that as /usr/local/Cellar/kustomize/2.0.1

The make sure you have them on your PATH:

```
cd /usr/local/bin
ln -s ../Cellar/kustomize/1.0.11/bin/kustomize
ln -s ../Cellar/kustomize/2.0.1/bin/kustomize kustomize2
```

You can find a new example app:

https://github.com/alexec/argocd-example-apps/tree/master/kustomize2-guestbook

# Notes on Kustomize 2

This appears to have a new security check, our example code produces this error:

```
Unable to generate manifests in kustomize2-guestbook: rpc error: code = Unknown desc = Error: rawResources failed to read Resources: Load from path ../guestbook/guestbook-ui-deployment.yaml failed: security; file '../guestbook/guestbook-ui-deployment.yaml' is not in or below '/private/var/folders/v5/0lllbd_s0ldg1dr4qxm7vw0scm2ffd/T/https:__github.com_alexec_argocd-example-apps_/kustomize2-guestbook
```

Also, please read https://github.com/kubernetes-sigs/kustomize/blob/master/docs/version2.0.0.md.
